data-prepper-plugins/otel-proto-common/src/test/java/com/amazon/dataprepper/plugins/otel/codec/OTelProtoCodecTest.java

```diff
@@ -0,0 +1,686 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.util.JsonFormat;
+import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceRequest;
+import io.opentelemetry.proto.common.v1.AnyValue;
+import io.opentelemetry.proto.common.v1.ArrayValue;
+import io.opentelemetry.proto.common.v1.InstrumentationLibrary;
+import io.opentelemetry.proto.common.v1.KeyValue;
+import io.opentelemetry.proto.common.v1.KeyValueList;
+import io.opentelemetry.proto.resource.v1.Resource;
+import io.opentelemetry.proto.trace.v1.InstrumentationLibrarySpans;
+import io.opentelemetry.proto.trace.v1.ResourceSpans;
+import io.opentelemetry.proto.trace.v1.Status;
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Random;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class OTelProtoCodecTest {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final Random RANDOM = new Random();
+    private static final String TEST_REQUEST_JSON_FILE = "test-request.json";
+    private static final String TEST_SPAN_EVENT_JSON_FILE = "test-span-event.json";
+
+    private final OTelProtoCodec.OTelProtoDecoder decoderUnderTest = new OTelProtoCodec.OTelProtoDecoder();
+    private final OTelProtoCodec.OTelProtoEncoder encoderUnderTest = new OTelProtoCodec.OTelProtoEncoder();
+
+    private byte[] getRandomBytes(int len) {
+        byte[] bytes = new byte[len];
+        RANDOM.nextBytes(bytes);
+        return bytes;
+    }
+
+    private Map<String, Object> returnMap(final String jsonStr) throws JsonProcessingException {
+        return (Map<String, Object>) OBJECT_MAPPER.readValue(jsonStr, Map.class);
+    }
+
+    private List<Object> returnList(final String jsonStr) throws JsonProcessingException {
+        return (List<Object>) OBJECT_MAPPER.readValue(jsonStr, List.class);
+    }
+
+    private ExportTraceServiceRequest buildExportTraceServiceRequestFromJsonFile(String requestJsonFileName) throws IOException {
```

Comment:
Nit: `requestJsonFileName` may be final.

Reply:
Oops! I missed your comment. Will make up in following PRs.